src/features/import/components/UploadFile.tsx

```diff
@@ -0,0 +1,177 @@
+import { useDropzone } from 'react-dropzone';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  IconButton,
+  Typography,
+} from '@mui/material';
+import { useCallback, useState } from 'react';
+
+import { ImportedFile } from '../utils/parseFile';
+import messageIds from '../l10n/messagesIds';
+import theme from 'theme';
+import { UploadFileOutlined } from '@mui/icons-material';
+import useImportedFile from '../hooks/useImportedFile';
+import { Msg, useMessages } from 'core/i18n';
+
+const UploadFile = () => {
+  const [uploadedFile, setUploadedFile] = useState<File | null>();
+  const [rawData, setRawData] = useState<ImportedFile>();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<boolean>(false);
+  const messages = useMessages(messageIds);
+  const { importData } = useImportedFile();
+
+  const onDrop = useCallback((acceptedFiles: File[]): void => {
+    acceptedFiles.map((file: File) => {
+      setLoading(true);
+      setUploadedFile(file);
+
+      importData(file)?.then((res) => {
+        setRawData(res);
+        setLoading(false);
+      });
```

Comment:
I think the loading state etc (all async stuff) should move into the hook, and the data should not be stored in state in this component, but in the store.